File: backend/Jawlah.Core/Entities/Issue.cs
using Jawlah.Core.Enums;

namespace Jawlah.Core.Entities;

public class Issue
{
    public int IssueId { get; set; }
    public int ReportedByUserId { get; set; }
    public int? ZoneId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public IssueType Type { get; set; }
    public IssueSeverity Severity { get; set; }
    public IssueStatus Status { get; set; }
    public double Latitude { get; set; }
    public double Longitude { get; set; }
    public string? LocationDescription { get; set; }
    public string? PhotoUrl { get; set; }
    public string? AdditionalPhotosJson { get; set; }
    public DateTime ReportedAt { get; set; }
    public DateTime? ResolvedAt { get; set; }
    public string? ResolutionNotes { get; set; }
    public int? ResolvedByUserId { get; set; }
    public DateTime EventTime { get; set; }
    public DateTime? SyncTime { get; set; }
    public bool IsSynced { get; set; }
    public int SyncVersion { get; set; }
    public User ReportedByUser { get; set; } = null!;
    public User? ResolvedByUser { get; set; }
    public Zone? Zone { get; set; }
}

File: web/src/types/issue.ts
// issue types

export type IssueType = "Infrastructure" | "Safety" | "Cleanliness" | "Equipment" | "Other";
export type IssueSeverity = "Low" | "Medium" | "High" | "Critical";
export type IssueStatus = "New" | "Forwarded" | "Resolved" | "InProgress" | "Closed" | "ConvertedToTask";

export type IssueResponse = {
  issueId: number;
  title: string;
  description: string;
  type: IssueType;
  severity: IssueSeverity;
  status: IssueStatus;
  reportedByUserId: number;
  reportedByName?: string;
  zoneId?: number;
  zoneName?: string;
  latitude: number;
  longitude: number;
  locationDescription?: string;
  photos: string[];
  reportedAt: string;
  resolvedAt?: string;
  resolutionNotes?: string;
  // Forwarding info
  forwardedToDepartmentId?: number;
  forwardedToDepartmentName?: string;
  forwardedAt?: string;
  forwardingNotes?: string;

  syncTime?: string;
  syncVersion: number;
  priority: string; // Alias for severity (for web dashboard compatibility)
};

export type ForwardIssueRequest = {
  departmentId: number;
  notes?: string;
};

export type UpdateIssueStatusRequest = {
  status: IssueStatus;
  resolutionNotes?: string;
};

export type CreateTaskFromIssueRequest = {
  assignedToUserId: number;
  priority: "Low" | "Medium" | "High" | "Urgent";
  dueDate?: string;
  requiresPhotoProof: boolean;
  taskType?: string;
};
